#!/usr/bin/env python
PACKAGE = 'ground_integrity_layer'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("enabled", bool_t, 0, "Whether to apply this plugin or not", True)
scale_enum = gen.enum([ gen.const("nano_E",      int_t, -9, "nano scale E-9"),
                       gen.const("micro_E",     int_t, -6, "micro scale E-6"),
                       gen.const("milli_E",      int_t, -3, "milli scale E-3"),
                       gen.const("unit_E", int_t, 0, "unit scale E0"),
                       gen.const("kilo_E", int_t, 3, "kilo scale E3"),
                       gen.const("mega_E", int_t, 6, "mega scale E6"),
                       gen.const("giga_E", int_t, 9, "giga scale E9")],
                     "An enum to set unit prefix")

gen.add('threshold_centre', double_t, 0, "Nominal radar value for no additonal cost - please set scale", 1.0, -999, 999)
gen.add("threshold_centre_scale", int_t, 0, "Scaling prefix for threshold centre", 0, -9, 9, edit_method=scale_enum)

gen.add('threshold_variance', double_t, 0, "Variance which defines a nominal range for radar values - please set scale", 200, 0, 999)
gen.add("threshold_variance_scale", int_t, 0, "Scaling prefix for threshold variance", 0, -9, 9, edit_method=scale_enum)

gen.add('use_lethal', bool_t, 0, "use lethal cost as max - not advised", False)

exit(gen.generate(PACKAGE, "ground_integrity_layer", "GroundIntegrityLayer"))
